substrate/frame/contracts/src/exec.rs

```diff
@@ -1001,11 +1004,13 @@ where
 					let contract = frame.contract_info.as_contract();
 					frame.nested_storage.enforce_subcall_limit(contract)?;
 
-					let caller = self.caller();
-					Contracts::<T>::deposit_event(
-						vec![T::Hashing::hash_of(&caller), T::Hashing::hash_of(&account_id)],
-						Event::Called { caller: caller.clone(), contract: account_id.clone() },
-					);
+					if !frame.read_only {
```

Comment:
Do we want to `deposit_event` for read_only call?

Reply:
`deposit_event` should be forbidden in read only mode. At least this is how it is on Ethereum.

Reply:
Sorry I misunderstood. The event should still be emitted.